### PR TITLE
Add new `Liftoff` bidder

### PR DESCRIPTION
**Pull request by marki1an**
Corresponding issue https://github.com/prebid/prebid-server-java/issues/2572

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/liftoff/model/LiftoffImpressionExt.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+@Builder(toBuilder = true)
+@Getter
+public class LiftoffImpressionExt {
+
+    ExtImpPrebid prebid;
+
+    ExtImpLiftoff bidder;
+
+    @JsonProperty("vungle")
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/liftoff/model/LiftoffImpressionExt.java:17]: No need for `@JsonProperty`

File: src/main/java/org/prebid/server/proto/openrtb/ext/request/liftoff/ExtImpLiftoff.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package org.prebid.server.proto.openrtb.ext.request.liftoff;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder(toBuilder = true)
+@Getter
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/proto/openrtb/ext/request/liftoff/ExtImpLiftoff.java:8]: @Value

There is only three params, can be done with `of` static method

File: src/test/java/org/prebid/server/bidder/liftoff/LiftoffBidderTest.java [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+    private LiftoffBidder target;
+
+    @Before
+    public void setUp() {
+        target = new LiftoffBidder(ENDPOINT_URL, currencyConversionService, jacksonMapper);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new RtbhouseBidder(
```
Comment (SerhiiNahornyi) [inline src/test/java/org/prebid/server/bidder/liftoff/LiftoffBidderTest.java:68]: Rtbhouse?

File: src/test/java/org/prebid/server/bidder/liftoff/LiftoffBidderTest.java [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getPayload)
+                .flatExtracting(BidRequest::getImp)
+                .satisfies(imps -> {
```
Comment (SerhiiNahornyi) [inline src/test/java/org/prebid/server/bidder/liftoff/LiftoffBidderTest.java:104]: Here can be added one more step with just imp extraction

File: src/test/java/org/prebid/server/bidder/liftoff/LiftoffBidderTest.java [hunk 5]
```diff
@@ -0,0 +1,354 @@
...
```
Comment (SerhiiNahornyi) [inline src/test/java/org/prebid/server/bidder/liftoff/LiftoffBidderTest.java:128]: Same here. Pls check similar occurrences

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/proto/openrtb/ext/request/liftoff/ExtImpLiftoff.java (hunk 2 above)
Comment (marki1an) [reply src/main/java/org/prebid/server/proto/openrtb/ext/request/liftoff/ExtImpLiftoff.java:8]: In our code we should update `ExtImpLiftoff` obj field.

File: src/main/java/org/prebid/server/bidder/liftoff/LiftoffBidder.java [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+                final LiftoffImpressionExt liftoffImpressionExt = resolveLiftoffImpressionExt(imp);
+                liftoffImpressionExtBidder = liftoffImpressionExt.getBidder();
+                final String buyeruid = ObjectUtil.getIfNotNull(bidRequest.getUser(), User::getBuyeruid);
+                final ExtImpLiftoff extImpLiftoff = createExtImpLiftoff(buyeruid, liftoffImpressionExtBidder);
+                final LiftoffImpressionExt impressionExt = liftoffImpressionExt.toBuilder()
+                        .vungle(extImpLiftoff)
+                        .build();
+                convertedExtImpLiftoff = resolveExtImpLiftoff(impressionExt);
+            } catch (IllegalArgumentException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/liftoff/LiftoffBidder.java:94]: Pls cover this case with unit test

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**